qiskit/transpiler/__init__.py

```diff
@@ -36,84 +36,165 @@
+    operations.  See :ref:`transpiler-preset-stage-init` for more detail.
+
+``layout``
+    Choose an initial mapping of virtual qubits to physical qubits, including expansion of the
+    circuit to contain explicit ancillas.  This stage sometimes subsumes ``routing``.  See
+    :ref:`transpiler-preset-stage-layout` for more detail.
+
+``routing``
+    Insert gates into the circuit to ensure it matches connectivity constraints of the
+    :class:`.Target`.  The inserted gates need not match the target ISA yet, so are often just
+    ``swap`` instructions.  This stage is sometimes omitted, when the ``layout`` stage handles its
+    job.  See :ref:`transpiler-preset-stage-routing` for more detail.
+
+``translation``
+    Convert all gates in the circuit to ones matching the :class:`Target`\\ s ISA.  See
+    :ref:`transpiler-preset-stage-translation` for more detail.
+
+``optimization``
+    Low-level, hardware-aware optimizations.  Unlike the abstract optimizations of the ``init``
+    stage, this stage acts on a physical circuit.  See :ref:`transpiler-preset-stage-optimization`
+    for more detail.
+
+``scheduling``
+    Insert :class:`~.circuit.Delay` instructions to make the wall-clock timing of a circuit
+    explicit.  This may also include hardware-aware online error reduction techniques such as
+    dynamical decoupling, which are dependent on knowing wall-clock timings.  See
+    :ref:`transpiler-preset-stage-scheduling` for more details.
+
+The preset transpiler pipelines can also be configured at a high level by setting an
+``optimization_level``.  This is in integer from 0 to 3 inclusive, indicating the relative effort to
+put into attempting to optimize the circuit for the hardware.  Level 0 disables all unnecessary
+optimizations; only transformations needed to make the circuit runnable at all will be present.  On
+the other end, level 3 enables a full barrage of optimization techniques, some of which can be very
+expensive in compilation time.  Similar to classical compilers, optimization level 3 is not always
+guaranteed to produce the best results.  Qiskit defaults to optimization level 2, as a trade-off
+between compilation time and the expected amount of optimization.
+
+The optimization level affects which implementations are used for a given stage by default, though
+this can be overridden by passing explicit ``<stage>_method="<choice>"`` arguments to
+:func:`.generate_preset_pass_manager`.
+
+.. note::
+
+    The preset pass managers almost always include stochastic, heuristic-based passes.  If you need
+    to ensure reproducibility of a compilation, pass a known integer to the ``seed_transpiler``
+    argument to the generator functions.
+
+    This stochasticity is because many of the problems the transpiler must solve are known to be
+    non-polynomial in complexity, and we need to ensure we finish the job in a workable amount of
```

Comment:
"we need to ensure" - can we reword to remove the "we"? "You need to ensure"... or "the job must finish in a workable amount of time"... something like this

Reply:
Changed to
> This stochasticity is because many of the problems the transpiler must solve are known to be non-polynomial in complexity, but transpilation must complete in a workable amount of time.